File: python/find-the-score-of-all-prefixes-of-an-array.py
class Solution:
    def findPrefixScore(self, nums: List[int]) -> List[int]:
        score = 0
        maximal = 0
        answer = []
        for num in nums:
            maximal = max(maximal, num)
            conversion_value = num + maximal
            score += conversion_value
            answer.append(score)
        return answer

File: go/find-the-score-of-all-prefixes-of-an-array.go
func findPrefixScore(nums []int) []int64 {
	answer := make([]int64, 0)
	score := int64(0)
	maximal := int64(0)
	for _, num := range nums {
		num := int64(num)
		if maximal < num {
			maximal = num
		}
		score += num + maximal
		answer = append(answer, score)
	}
	return answer
}
